### PR TITLE
Get rid of KeyErrors in to_line_graph 

### PR DESCRIPTION
**Pull request by pgberlureau**
Use edge_label_dict.values() as nodes and get edge1 and edge2 from edge_label_dict.keys() to avoid KeyError.
(Fix issue #557)

### CODE REVIEW

File: xgi/convert/line_graph.py [hunk 1]
```diff
@@ -55,16 +55,16 @@ def to_line_graph(H, s=1, weights=None):
 
     edge_label_dict = {tuple(edge): index for index, edge in H._edge.items()}
 
-    LG.add_nodes_from(H.edges)
+    LG.add_nodes_from(edge_label_dict.values())
 
-    for edge1, edge2 in combinations(H.edges.members(), 2):
+    for edge1, edge2 in combinations(edge_label_dict.keys(), 2):
```
Comment (nwlandry) [inline xgi/convert/line_graph.py:60]: If you just use `edge_label_dict` it will use the keys by default.

**Comment by nwlandry** [comment]
@pgberlureau thanks so much for finding this bug and for your contribution! Two quick questions:
1. Can you add a corresponding unit test in `tests/convert/test_line_graph.py` called `test_557` so that we don't break this in the future?
2. I realized that we are eliminating duplicate edges by making the tuples of the edges the keys of the label dict. Perhaps we should do something like this:
```python
for e1, e2 in combinations(H._edge, 2):
    edge_intersect = len(H._edge[e1].intersection(H._edge[e2]))
    if edge_intersect >= s:
        LG.add_edge(e1, e2)
```

**Comment by nwlandry** [comment]
I saw that the unit tests are failing due to an unrelated issue which I will fix. Thanks again!!

**Comment by pgberlureau** [comment]
Hey,

1. yes i can add the test, no problem :)
2. You are right, only thing is that in the obtained nx graph, nodes are now indices and the original corresponding hyperedges are  attributes of those nodes. This is not a problem to me but is this really the desired behaviour ?

**Comment by nwlandry** [comment]
> Hey,
> 
> 1. yes i can add the test, no problem :)
> 2. You are right, only thing is that in the obtained nx graph, nodes are now indices and the original corresponding hyperedges are  attributes of those nodes. This is not a problem to me but is this really the desired behaviour ? (Or I didn't got what you meant sorry)

What I was realizing is that using tuples may be unexpectedly buggy. Because the order of `set` is not guaranteed, when we cast to tuples, we may be getting different permutations of the edges, e.g., `(1, 2, 3)` and `(3, 1, 2)`. I propose that we fix this using the code that I suggested, or, at the very least, putting a `sorted` statement inside the casting to tuple.

My worry about using the edges as keys is that the line graph will be the same with or without multiedges, so if a hypergraph is not a simple hypergraph, then we're not getting the accurate linegraph.

**Review by nwlandry — APPROVED** [review]
This looks awesome, @pgberlureau! Thanks so much! Let me know if you're done, and if so, I'll merge.

**Comment by pgberlureau** [comment]
I have nothing to add, thank you for your work !

**Pull request merged**